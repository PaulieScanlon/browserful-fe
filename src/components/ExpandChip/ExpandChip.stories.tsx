import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { State, Store } from '@sambego/storybook-state';

import { ExpandChip } from './ExpandChip';

const store = new Store({
  isExpanded: true
});

const stories = storiesOf('ExpandChip', module);

const onClick = (event, store) => {
  store.set({ isExpanded: !store.get('isExpanded') });
  action('onExpandClick')(
    event.currentTarget,
    `isExpanded: ${store.get('isExpanded')}`
  );
};

stories.add(
  'usage',
  withInfo(
    'ExpandChip is used in BrowserCard and contains an onClick to expand the BrowserCard. The isExpaned prop controls the rotation of the chevron'
  )(() => (
    <State store={store}>
      <ExpandChip isExpanded={true} onClick={event => onClick(event, store)} />
    </State>
  ))
);
