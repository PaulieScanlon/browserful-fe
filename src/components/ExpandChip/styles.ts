import styled from 'react-emotion';
import { colours } from '../../theme';

interface IProps {
  isExpaned: boolean;
}

export const ExpandChipWrapper = styled.span<IProps>(
  {
    label: 'expand-chip-wrapper',
    display: 'inline-flex',
    borderRadius: '50%',
    backgroundColor: colours.offWhite,
    cursor: 'pointer',
    svg: {
      fill: colours.greyMid
    }
  },
  ({ isExpaned }) => ({
    transform: isExpaned ? `rotate(0deg)` : `rotate(180deg)`
  })
);
