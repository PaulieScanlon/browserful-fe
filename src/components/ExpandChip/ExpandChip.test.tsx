import * as React from 'react';
import { shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';
expect.extend(createMatchers(emotion));

import { ExpandChip } from './ExpandChip';

const onClick = event => {};

const shallowDefault = props => {
  return shallow(<ExpandChip {...props} onClick={event => onClick(event)} />);
};

describe('<ExpandChip>', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault({ isExpaned: true });
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls function onClick', () => {
    const testOnClick = jest.fn();
    const wrapper = shallow(
      <ExpandChip isExpanded={true} onClick={event => testOnClick(event)} />
    );
    wrapper.simulate('click');
    expect(testOnClick).toHaveBeenCalledTimes(1);
  });
  it('has correct styles for isExpanded:true', () => {
    const wrapper = shallowDefault({ isExpanded: true });
    expect(wrapper).toHaveStyleRule('transform', 'rotate(0deg)');
  });
  it('has correct styles for isExpanded:false', () => {
    const wrapper = shallowDefault({ isExpanded: false });
    expect(wrapper).toHaveStyleRule('transform', 'rotate(180deg)');
  });
});
