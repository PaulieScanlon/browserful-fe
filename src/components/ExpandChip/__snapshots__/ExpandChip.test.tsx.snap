// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandChip> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpandChip
    isExpaned={true}
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <svg
        height="20"
        version="1.1"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          expand-chip
        </title>
        <path
          d="M15.484 12.452c-0.436 0.446-1.043 0.481-1.576 0l-3.908-3.747-3.908 3.747c-0.533 0.481-1.141 0.446-1.574 0-0.436-0.445-0.408-1.197 0-1.615 0.406-0.418 4.695-4.502 4.695-4.502 0.217-0.223 0.502-0.335 0.787-0.335s0.57 0.112 0.789 0.335c0 0 4.287 4.084 4.695 4.502s0.436 1.17 0 1.615z"
        />
      </svg>,
      "isExpaned": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <title>
            expand-chip
          </title>,
          <path
            d="M15.484 12.452c-0.436 0.446-1.043 0.481-1.576 0l-3.908-3.747-3.908 3.747c-0.533 0.481-1.141 0.446-1.574 0-0.436-0.445-0.408-1.197 0-1.615 0.406-0.418 4.695-4.502 4.695-4.502 0.217-0.223 0.502-0.335 0.787-0.335s0.57 0.112 0.789 0.335c0 0 4.287 4.084 4.695 4.502s0.436 1.17 0 1.615z"
          />,
        ],
        "height": "20",
        "version": "1.1",
        "viewBox": "0 0 20 20",
        "width": "20",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "expand-chip",
          },
          "ref": null,
          "rendered": "expand-chip",
          "type": "title",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M15.484 12.452c-0.436 0.446-1.043 0.481-1.576 0l-3.908-3.747-3.908 3.747c-0.533 0.481-1.141 0.446-1.574 0-0.436-0.445-0.408-1.197 0-1.615 0.406-0.418 4.695-4.502 4.695-4.502 0.217-0.223 0.502-0.335 0.787-0.335s0.57 0.112 0.789 0.335c0 0 4.287 4.084 4.695 4.502s0.436 1.17 0 1.615z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <svg
          height="20"
          version="1.1"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            expand-chip
          </title>
          <path
            d="M15.484 12.452c-0.436 0.446-1.043 0.481-1.576 0l-3.908-3.747-3.908 3.747c-0.533 0.481-1.141 0.446-1.574 0-0.436-0.445-0.408-1.197 0-1.615 0.406-0.418 4.695-4.502 4.695-4.502 0.217-0.223 0.502-0.335 0.787-0.335s0.57 0.112 0.789 0.335c0 0 4.287 4.084 4.695 4.502s0.436 1.17 0 1.615z"
          />
        </svg>,
        "isExpaned": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <title>
              expand-chip
            </title>,
            <path
              d="M15.484 12.452c-0.436 0.446-1.043 0.481-1.576 0l-3.908-3.747-3.908 3.747c-0.533 0.481-1.141 0.446-1.574 0-0.436-0.445-0.408-1.197 0-1.615 0.406-0.418 4.695-4.502 4.695-4.502 0.217-0.223 0.502-0.335 0.787-0.335s0.57 0.112 0.789 0.335c0 0 4.287 4.084 4.695 4.502s0.436 1.17 0 1.615z"
            />,
          ],
          "height": "20",
          "version": "1.1",
          "viewBox": "0 0 20 20",
          "width": "20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "expand-chip",
            },
            "ref": null,
            "rendered": "expand-chip",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M15.484 12.452c-0.436 0.446-1.043 0.481-1.576 0l-3.908-3.747-3.908 3.747c-0.533 0.481-1.141 0.446-1.574 0-0.436-0.445-0.408-1.197 0-1.615 0.406-0.418 4.695-4.502 4.695-4.502 0.217-0.223 0.502-0.335 0.787-0.335s0.57 0.112 0.789 0.335c0 0 4.287 4.084 4.695 4.502s0.436 1.17 0 1.615z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
